src/plugins/app-version-checker-plugin.ts

```diff
@@ -0,0 +1,35 @@
+import Hapi from '@hapi/hapi';
+import {compareVersion} from '../utils/compare-version';
+
+interface Options {}
+
+const plugin: Hapi.Plugin<Options> = {
+  dependencies: 'atb-headers',
+  name: 'app-version-checker-plugin',
+  register: (server) => {
+    server.ext('onPreHandler', (request, h) => {
+      const minAppVersion = process.env.MIN_APP_VERSION;
+      if (isAppVersionTooLow(request, minAppVersion)) {
+        return h
+          .response({error: `Required client app version: ${minAppVersion}`})
+          .code(426)
```

Comment:
We shouldn't return 426 here, as it indicates that the HTTP _protocol_ needs to be upgraded, ie. if the client uses HTTP 1.1 but the server requires 2.0. It is also required that you respond with an `Upgrade` header saying which protocols are allowed. See [RFC 7231, section 6.5.15](https://datatracker.ietf.org/doc/html/rfc7231#section-6.5.15) for more details.

We could go for 406 (Not Acceptable), as it seems to indicate that it's not acceptable for the user-agent (which is what we are doing). See [RFC 7231, setion 6.5.6](https://datatracker.ietf.org/doc/html/rfc7231#section-6.5.6).